src/bun.js/node/hdr_histogram.zig

```diff
@@ -0,0 +1,503 @@
+        }
+
+        if (options.lowest_trackable_value < 1) {
+            return error.InvalidLowestTrackableValue;
+        }
+
+        // Calculate derived values for efficient bucketing
+        // HDR Histogram is optimized for writes using bitwise operations and bit shifting, so we precalculate bitmasks and other helpful values
+
+        // upper value bound of each bucket
+        const largest_value_in_bucket = 2 * std.math.pow(u64, 10, options.significant_figures);
+        const log2largest_value = std.math.log2(@as(f64, @floatFromInt(largest_value_in_bucket)));
+        const sub_bucket_count_magnitude: u8 = @intFromFloat(@ceil(log2largest_value)); // bits required to represent largest value, rounded up
+
+        const sub_bucket_count = std.math.pow(u64, 2, sub_bucket_count_magnitude); // actual quantity of sub-buckets per bucket, defaults to 2048
+        const sub_bucket_half_count = sub_bucket_count / 2;
+        const sub_bucket_half_count_magnitude: u6 = @truncate(sub_bucket_count_magnitude - 1);
+
+        // lower bound of each bucket
+        const log2lowest_value = std.math.log2(@as(f64, @floatFromInt(options.lowest_trackable_value)));
+        const unit_magnitude = @as(u8, @intFromFloat(std.math.floor(log2lowest_value)));
+
+        // represent this as a mask of 1s for efficient bitwise operations
+        const sub_bucket_mask = (sub_bucket_count - 1) * std.math.pow(u64, 2, unit_magnitude);
+
+        // add more buckets if we need to track higher values
+        var bucket_count: u32 = 1;
+        var smallest_untrackable_value = sub_bucket_count * std.math.pow(u64, 2, unit_magnitude);
+        while (smallest_untrackable_value <= options.highest_trackable_value) {
+            if (smallest_untrackable_value > std.math.maxInt(u64) / 2) {
+                // next step would overflow, so we just increment the bucket count and break
+                bucket_count += 1;
+                break;
+            }
+            smallest_untrackable_value = 2 * smallest_untrackable_value;
+            bucket_count += 1;
+        }
+        const counts_len = (bucket_count + 1) * sub_bucket_half_count;
+        const counts = try allocator.alloc(u64, counts_len);
+        for (0..counts_len) |i| {
+            counts[i] = 0;
+        }
+
+        return This{
+            .allocator = allocator,
+            .lowest_trackable_value = options.lowest_trackable_value,
+            .highest_trackable_value = options.highest_trackable_value,
+            .significant_figures = options.significant_figures,
+            .sub_bucket_count = sub_bucket_count,
+            .sub_bucket_half_count = sub_bucket_half_count,
+            .sub_bucket_half_count_magnitude = sub_bucket_half_count_magnitude,
+            .unit_magnitude = unit_magnitude,
+            .sub_bucket_mask = sub_bucket_mask,
+            .bucket_count = bucket_count,
+            .counts = counts,
+            .total_count = 0,
+            .min = 9223372036854776000,
```

Comment:
should this start as `highest_trackable_value`? or `std.math.maxInt(u64)`?

Reply:
std.math.maxInt is probably better choice, whole point is to guarantee it gets overridden on next record